File: src/Moryx.ControlSystem.Processes.Endpoints/Models/JobProcessModel.cs
// Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;

namespace Moryx.ControlSystem.Processes.Endpoints;

[DataContract]
public class JobProcessModel
{
    [DataMember]
    public long Id { get; set; }

    [DataMember]
    public long RecipeId { get; set; }

    [DataMember]
    public ProcessProgress State { get; set; }

    [DataMember]
    public bool Rework { get; set; }

    [DataMember]
    public DateTime Started { get; set; }

    [DataMember]
    public DateTime Completed { get; set; }

    [DataMember]
    public ProcessActivityModel[] Activities { get; set; }

    [DataMember]
    public bool IsRunning { get; set; }
}

File: src/Moryx.ControlSystem.ProcessEngine.Web/src/app/api/models/Moryx/ControlSystem/Processes/Endpoints/job-process-model.ts
/* tslint:disable */
/* eslint-disable */
import { ProcessActivityModel as MoryxControlSystemProcessesEndpointsProcessActivityModel } from '../../../../../models/Moryx/ControlSystem/Processes/Endpoints/process-activity-model';
import { ProcessProgress as MoryxControlSystemProcessesProcessProgress } from '../../../../../models/Moryx/ControlSystem/Processes/process-progress';
export interface JobProcessModel {
  activities?: Array<MoryxControlSystemProcessesEndpointsProcessActivityModel> | null;
  completed?: string;
  id?: number;
  isRunning?: boolean;
  recipeId?: number;
  rework?: boolean;
  started?: string;
  state?: MoryxControlSystemProcessesProcessProgress;
}
